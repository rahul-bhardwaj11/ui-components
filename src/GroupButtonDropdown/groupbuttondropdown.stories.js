import React from 'react';

import { storiesOf } from '@storybook/react';
import PropTypes from 'prop-types';
import GroupButtonDropdown from './index';
import { withInfo } from '@storybook/addon-info';
import { withKnobs, object, text } from '@storybook/addon-knobs';

const stories = storiesOf('GroupButtonDropdown', module);
stories.addDecorator(withKnobs);

const options = [
  {
    key: '1',
    content:
      'First Item First ItemFirst ItemFirst ItemFirst ItemFirst ItemFirst ItemFirst ItemFirst Item'
  },
  { key: '2', content: 'Second Item' },
  { key: '3', content: 'Third Item' }
];

class SomeReactComponent extends React.Component {
  static propTypes = {
    content: PropTypes.any
  };

  render() {
    let { content } = this.props;
    return <div style={{ left: -50 }}>{content}</div>;
  }
}

stories
  .add(
    'Default GroupButtonDropdown',
    withInfo('Basic usage of the GroupButtonDropdown')(() => (
      <GroupButtonDropdown
        options={object('options', options)}
        label={text('label', 'Add')}
      />
    ))
  )
  .add(
    'GroupButtonDropdown with any HTML children',
    withInfo('GroupButtonDropdown with any HTML children')(() => (
      <GroupButtonDropdown
        options={
          <SomeReactComponent content="This is child component for dropdown" />
        }
        label={text('label', 'Add')}
      />
    ))
  );
