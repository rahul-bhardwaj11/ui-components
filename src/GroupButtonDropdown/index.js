import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Dropdown from '../Dropdown';
import Button from '../Button';
import styled from 'styled-components';
import Icon from '../Icon';

const MtGroupDropdown = styled.div`
  .ant-dropdown {
    left: 0 !important;
  }
  .ant-dropdown-trigger {
    .ant-btn-primary {
      span {
        font-size: 8px;
        margin-top: 7px;
        display: inline-block;
        vertical-align: top;
      }
    }
    &.dropdownBtn {
      padding: 0px 10px;
      border-bottom-left-radius: 0;
      border-top-left-radius: 0;
      border-left: 1px solid white;
      &:hover,
      &:focus {
        border-left: 1px solid white;
      }
    }
  }
  .ant-dropdown-menu-light {
    border: 1px solid rgba(205, 210, 217, 0.5);
    border-radius: 4px;
    background-color: white;
    box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.08);
  }
  .downFillCaretIcon {
    font-size: 8px;
  }
  .labelBtn {
    display: inline-block;
    padding: 0px 10px;
    border-bottom-right-radius: 0;
    border-top-right-radius: 0;
  }
`;

class GroupButtonDropdown extends Component {
  static propTypes = {
    options: PropTypes.any.isRequired,
    trigger: PropTypes.string,
    onSelect: PropTypes.func,
    label: PropTypes.string
  };

  static defaultProps = {
    trigger: 'click',
    onSelect: () => {}
  };

  render() {
    let { label, ...rest } = this.props;
    return (
      <MtGroupDropdown>
        <Button className="labelBtn" onClick={this.props.onSelect}>
          {label}
        </Button>
        <Dropdown {...rest}>
          <Button className="dropdownBtn">
            <Icon type="down_fillcaret" className="downFillCaretIcon" />
          </Button>
        </Dropdown>
      </MtGroupDropdown>
    );
  }
}
export default GroupButtonDropdown;
