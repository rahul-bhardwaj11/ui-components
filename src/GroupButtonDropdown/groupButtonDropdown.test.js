import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import GroupButtonDropdown from './index';

describe('GroupButtonDropdown', () => {
  const options = [
    { key: '1', content: 'First Item' },
    { key: '2', content: 'Second Item' },
    { key: '3', content: 'Third Item' }
  ];
  it('matches default snapshot', () => {
    const tree = renderer
      .create(<GroupButtonDropdown options={options} label="ADD" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should handle click', () => {
    const clickSpy = jest.fn();
    const groupButtonDropdown = shallow(
      <GroupButtonDropdown
        options={options}
        trigger="click"
        onClick={clickSpy}
      />
    );
    groupButtonDropdown.find('Dropdown').simulate('click', { key: '1' });
    expect(clickSpy).toHaveBeenCalled();
  });
});
