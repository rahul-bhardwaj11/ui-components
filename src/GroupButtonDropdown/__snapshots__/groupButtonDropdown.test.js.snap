// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GroupButtonDropdown matches default snapshot 1`] = `
<div
  className="sc-htpNat cvEUIa"
>
  <div
    className="sc-bwzfXH cujgKv"
  >
    <button
      className="ant-btn eeee ant-btn-primary"
      disabled={false}
      onClick={[Function]}
      style={
        Object {
          "borderBottomRightRadius": 0,
          "borderTopRightRadius": 0,
          "display": "inline-block",
          "padding": "0px 10px",
        }
      }
      type="button"
    >
      <span>
        ADD
      </span>
    </button>
  </div>
  <div
    className="ant-dropdown-trigger"
    disabled={undefined}
    onBlur={undefined}
    onClick={[Function]}
    onContextMenu={undefined}
    onFocus={undefined}
    onMouseDown={[Function]}
    onMouseEnter={undefined}
    onMouseLeave={undefined}
    onTouchStart={[Function]}
  >
     
    <div
      style={
        Object {
          "display": "inline-block",
        }
      }
    >
      <div
        className="sc-bwzfXH cujgKv"
      >
        <button
          className="ant-btn ant-btn-primary"
          disabled={false}
          onClick={[Function]}
          style={
            Object {
              "borderBottomLeftRadius": 0,
              "borderLeft": "1px solid",
              "borderTopLeftRadius": 0,
              "padding": "0px 10px",
            }
          }
          type="button"
        >
          <span>
            â–¼
          </span>
        </button>
      </div>
    </div>
     
  </div>
</div>
`;
